Wikipedia/Code/SearchLanguagesBarViewController.swift

```diff
@@ -0,0 +1,145 @@
+import Foundation
+
+@objc protocol SearchLanguagesBarViewControllerDelegate: class {
+    func searchLanguagesBarViewController(controller: SearchLanguagesBarViewController, didChangeCurrentlySelectedSearchLanguage language: MWKLanguageLink)
+}
+
+class SearchLanguagesBarViewController: UIViewController, WMFPreferredLanguagesViewControllerDelegate {
+    weak var delegate: SearchLanguagesBarViewControllerDelegate?
+    
+    @IBOutlet private var languageButtons: [UIButton] = []
+    @IBOutlet private var otherLanguagesButton: UIButton?
+    @IBOutlet private var heightConstraint: NSLayoutConstraint?
+    
+    private var previousFirstLanguage: MWKLanguageLink?
+    private var hidden: Bool = false {
+        didSet {
+            if(hidden){
+                heightConstraint!.constant = 0
+                view.hidden = true
+            }else{
+                heightConstraint!.constant = 44
+                view.hidden = false
+            }
+        }
+    }
+
+    private var _currentlySelectedSearchLanguage: MWKLanguageLink?
+    private(set) var currentlySelectedSearchLanguage: MWKLanguageLink? {
+        get {
+            return _currentlySelectedSearchLanguage
```

Comment:
Introducing extra state - `currentlySelectedSearchLanguage` should never be different than `NSUserDefaults.wmf_userDefaults().wmf_currentSearchLanguageDomain`. This should just return/set that value.